crates/bevy_ecs/src/component.rs

```diff
@@ -12,26 +12,85 @@ use std::{
     any::{Any, TypeId},
 };
 
-/// A component is data associated with an [`Entity`](crate::entity::Entity). Each entity can have
-/// multiple different types of components, but only one of them per type.
+// TODO: The content in the section “Implementing the trait for foreign types” may be more appropriate
+// for the Bevy book. However, until the change gets published, it's better to keep it there. Once the
+// book includes usage of the pattern, the section can be replaced with a link to the appropriate
+// section of the book.
+
+/// A data type that can be associated to an [entity].
+///
+/// `Component` is a [derivable trait]:
+/// this means that a data type can implement it by simply prepending a `#[derive(Component)]` attribute
+/// to it. It is necessary though that it satisfies the `Send + Sync + 'static` bounds.
+///
+/// In the context of a [`World`], a component instance must belong to a single entity.
```

Comment:
I'm really not sure what the intent is here.

Reply:
Ah, I see. This is a useful thing to bear in mind, but the wording was very unclear to me.

IMO this covers 99% of what users need to know.

> Each component is owned by exactly one entity; data is not deduplicated in any fashion.

The fact that you can generate more objects of the same type when they're not being used as components is obvious to me, and confusing and irrelevant to beginners.

Reply:
I like your version. A question: since this explains the relationship between components and entities, should this be placed in the second paragraph, right before the “`Component` is a derivable trait [...]” sentence?

Reply:
I definitely like that better.